pkg/apply/event/event.go

```diff
@@ -261,15 +261,13 @@ type PruneEvent struct {
 	Identifier object.ObjMetadata
 	Operation  PruneEventOperation
 	Object     *unstructured.Unstructured
-	// If prune is skipped, this reason string explains why
-	Reason string
-	Error  error
+	Error      error
```

Comment:
This change will break the cases when pruning objects are skipped, but the while case should pass. For example, the lifecycle annotation for preventing deletion. 

I think we need to keep the `Reason` field.

Reply:
> This change will break the cases when pruning objects are skipped, but the while case should pass.

The skip will be returned as an error, but errors don't stop the task or the task queue. So it should just continue the same way it does now, right?

Callers do need to be modifier to allow for this, like I did in the printer code.

Reply:
Yes, the callers need to to consume the prune events differently. Are the errors related to dependency all with specific error types?

Reply:
> Are the errors related to dependency all with specific error types?

That's part of this PR. I added the following:
- CurrentUIDFilter:
  - filter.ApplyPreventedDeletionError
- DependencyFilter
  - filter.DependencyPreventedActuationError
  - filter.DependencyActuationMismatchError
- InventoryPolicyApplyFilter & InventoryPolicyPruneFilter:
  - inventory.PolicyPreventedActuationError
- LocalNamespacesFilter:
  - filter.NamespaceInUseError
- PreventRemoveFilter:
  - filter.AnnotationPreventedDeletionError

Reply:
Yes, each of these has an error string explaining why, mostly for kpt tho. Config Sync likes to catch them and translate them into specific CS error codes.